volume/mounts/mounts.go

```diff
@@ -202,6 +202,20 @@ func (m *MountPoint) Setup(ctx context.Context, mountLabel string, rootIDs idtoo
 		return volumePath, clean, nil
 	}
 
+	if m.Type == mounttypes.TypeImage {
+		if m.Spec.ImageOptions != nil && m.Spec.ImageOptions.Subpath != "" {
+			subpath := m.Spec.ImageOptions.Subpath
+
+			safePath, err := safepath.Join(ctx, m.Source, subpath)
+			if err != nil {
+				return "", noCleanup, err
+			}
+			m.safePaths = append(m.safePaths, safePath)
+			log.G(ctx).Debugf("mounting (%s|%s) via %s", m.Source, subpath, safePath.Path())
+			return safePath.Path(), noCleanup, nil
```

Comment:
Note: we don't necessarily need the safepath there as the image is already immutable so there won't be a way for the attacker to exploit the TOCTOU with symlink exchange.

This could be simplified to just evaluating symlinks and making sure it's inside `m.Source`.

Reply:
If we intend to allow writable mount in the future, wouldn't this make it easier then?

Reply:
Also, wouldn't there a vector if the image being mounted has a malicious symlink?

Reply:
> If we intend to allow writable mount in the future, wouldn't this make it easier then?

Each container should get its own RW layer anyway, so there's only one container ever that can mutate the filesystem, but that container doesn't even exist at this point as we're preparing a request to create it.

So it's definitely easier than with volumes, where each can be mounted to a multiple running containers.

Reply:
> Also, wouldn't there a vector if the image being mounted has a malicious symlink?

If we resolve the symlink and verify it doesn't escape the image root dir, then we're good. It's enough, because the image is read-only so it won't be mutated.

Reply:
We can leave the safepath though, as it takes care of the above (resolve symlink and ensure it doesn't escape), but see the suggestion below (we need to return `safePath.Close`)